core/src/block_sync.rs

```diff
@@ -242,22 +246,6 @@ pub mod message {
         }
     }
 
-    /// Message variant to send our current latest block
-    #[derive(Io, Decode, Encode, Debug, Clone)]
-    pub struct LatestBlock {
-        /// Block hash
-        pub hash: HashOf<VersionedCommittedBlock>,
-        /// Peer id
-        pub peer_id: PeerId,
-    }
-
-    impl LatestBlock {
-        /// Default constructor
-        pub const fn new(hash: HashOf<VersionedCommittedBlock>, peer_id: PeerId) -> Self {
-            Self { hash, peer_id }
-        }
-    }
```

Comment:
We still need this message. As mentioned in the task description peer should request latest block hashes from selected number of peers, then compare with its own latest block hash, after that in case of difference try to synchronize.

Reply:
@eadventurous it seemed redundant to me to. I thought peer requesting latest blocks from other peers would just send it's latest block hash in `GetBlocksAfter` and other peers would respond with blocks after the latest block hash peer had sent

Reply:
why wouldn't peers that provide latest blocks be put in the position to compare and respond if required?

Reply:
because, otherwise, we are introducing another roundtrip of messaging

Reply:
In the current PR, considering the communication between two nodes, there are 2 messages sent in total. In the case you are proposing there are 4 messages sent in total. Previously, there were 3 messages sent

Reply:
Then there might be a situation:
1. New peer joins the network - it has 0 blocks
2. Network is already operating for a while and has 100 GB of blocks
3. Peer asks a set of N other randomly selected peers to give blocks after initial zero hash.
4. All N peers start to send 100 GB of blocks to it.

This seems to both create an extensive load on existing peers and on the new peer, with a lot of redundant bytes being sent.

Reply:
Synchronization will be N times slower also in this case.

Reply:
I see, payload is an issue, tnx